src/Adapters/Storage/Postgres/PostgresStorageAdapter.js

```diff
@@ -746,40 +747,37 @@ export class PostgresStorageAdapter {
   addFieldIfNotExists(className, fieldName, type) {
     // TODO: Must be revised for invalid logic...
     debug('addFieldIfNotExists', {className, fieldName, type});
-    return this._client.tx('add-field-if-not-exists', t => {
-      let promise = Promise.resolve();
+    const self = this;
+    return this._client.tx('add-field-if-not-exists', function * (t) {
       if (type.type !== 'Relation') {
-        promise = t.none('ALTER TABLE $<className:name> ADD COLUMN $<fieldName:name> $<postgresType:raw>', {
-          className,
-          fieldName,
-          postgresType: parseTypeToPostgresType(type)
-        })
-          .catch(error => {
+        try {
+          yield t.none('ALTER TABLE $<className:name> ADD COLUMN $<fieldName:name> $<postgresType:raw>', {
+            className,
+            fieldName,
+            postgresType: parseTypeToPostgresType(type)
+          });
+        } catch(error) {
             if (error.code === PostgresRelationDoesNotExistError) {
-              return this.createClass(className, {fields: {[fieldName]: type}})
-            } else if (error.code === PostgresDuplicateColumnError) {
-            // Column already exists, created by other request. Carry on to
-            // See if it's the right type.
-            } else {
-              throw error;
+              return yield self.createClass(className, {fields: {[fieldName]: type}}, t);
             }
-          })
+            if (error.code === PostgresDuplicateColumnError) {
+              // Column already exists, created by other request. Carry on to
+              // See if it's the right type.
```

Comment:
We should return here and not propagate the error to match original implementation. Otherwise, remove altogether.

Reply:
But I did match the original implementation. Was it somehow wrong?

Reply:
In the original implementation, the throw is in an else statement, so never executed if it enters the DuplicateColumnError check